src/authenticatorservice.rs

```diff
@@ -214,68 +147,7 @@ impl AuthenticatorService {
     pub fn register(
         &mut self,
         timeout: u64,
-        ctap_args: RegisterArgs,
-        status: Sender<crate::StatusUpdate>,
-        callback: StateCallback<crate::Result<crate::RegisterResult>>,
-    ) -> crate::Result<()> {
-        match ctap_args {
-            RegisterArgs::CTAP1(a) => self.register_ctap1(timeout, a, status, callback),
-            RegisterArgs::CTAP2(a) => self.register_ctap2(timeout, a, status, callback),
-        }
-    }
-
-    fn register_ctap1(
-        &mut self,
-        timeout: u64,
-        args: RegisterArgsCtap1,
-        status: Sender<crate::StatusUpdate>,
-        callback: StateCallback<crate::Result<crate::RegisterResult>>,
-    ) -> crate::Result<()> {
-        if args.challenge.len() != PARAMETER_SIZE || args.application.len() != PARAMETER_SIZE {
-            return Err(AuthenticatorError::InvalidRelyingPartyInput);
-        }
-
-        for key_handle in &args.key_handles {
-            if key_handle.credential.len() >= 256 {
```

Comment:
About removing those checks:
It seems, that `u2f_is_keyhandle_valid()` is handling these as well. We will not error out, though, but silently ignore those that are not valid (which I think is ok).
**But** `u2f_is_keyhandle_valid()` has an off-by-one error in that check: It checks for `> 256`, when it should be `>= 256`.

Reply:
Good catch!